admin/autoupdater/files/usr/lib/lua/autoupdater/util.lua

```diff
@@ -21,7 +36,41 @@ function popen(command)
     inr:close()
     inw:close()
 
-    nixio.exec('/bin/sh', '-c', command)
+    nixio.execp(...)
+    os.exit(127)
+  end
+end
+
+
+-- Executes all executable files in a directory
+function run_dir(dir)
+  local function is_ok(entry)
+    if entry:sub(1, 1) == '.' then
+      return false
+    end
+
+    local file = dir .. '/' .. entry
+    if fs.stat(file, 'type') ~= 'reg' then
```

Comment:
fs.stat gives the information about the file linked to, to get the information about the link itself, fs.lstat must be used.